### PR TITLE
chore: create separate module for each component

### PR DESCRIPTION
**Pull request by rayhan-ab**
no new features. this PR is done to decrease AP bundle size by separating each component instead of bundling everything into one single huge bundle. it allows AP to exclude unused components of ui-lib from its bundle. if a component is used, it will be downloaded lazily, only when it is needed.

note: ignore the build folder. what matters is the files outside it

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -88,5 +90,10 @@
   },
   "resolutions": {
     "react": "17.0.2"
-  }
+  },
+  "sideEffects": [
```
Comment (rayhan-ab) [inline package.json:94]: without this, ui-lib is not tree-shakeable. even if components are separated, they are all included in AP main bundle.
if this is set to false, ui-lib is tree-shakeable, but it won't have any css

File: rollup.config.js [hunk 2]
```diff
@@ -13,24 +13,53 @@ import typescript from "@rollup/plugin-typescript";
...
-      file: packageJson.module,
+      dir: "build",
       format: "esm",
-      sourcemap: true,
       assetFileNames: "[name]-[hash][extname]",
     },
   ],
   plugins: [
     peerDepsExternal(),
+    replace({
```
Comment (rayhan-ab) [inline rollup.config.js:56]: it reduces overall bundle size and build time

File: rollup.config.js [hunk 3]
```diff
@@ -13,24 +13,53 @@ import typescript from "@rollup/plugin-typescript";
...
+      "process.env.NODE_ENV": JSON.stringify("production"),
+    }),
     resolve(),
     commonjs(),
     typescript({ tsconfig: "./tsconfig.json", exclude: "**/*.stories.tsx" }),
+    multiInput(),
```
Comment (rayhan-ab) [inline rollup.config.js:62]: to retain folder structure inside build folder

**Comment by rayhan-ab** [comment]
i don't plan to make it like common-utils. we run build when it is installed as it will slow the AP pipeline down

File: rollup.config.js [hunk 4]
```diff
@@ -13,24 +13,53 @@ import typescript from "@rollup/plugin-typescript";
...
 
-import packageJson from "./package.json";
+function findFilesInDirectory(startPath, fileExtensionFilters) {
+  let results = [];
+  if (!fs.existsSync(startPath)) {
+    return;
+  }
+
+  const files = fs.readdirSync(startPath);
+  const forbiddenExtension = [".d.ts", ".stories.tsx"];
```
Comment (aziz-zen-ab) [inline rollup.config.js:28]: `forbiddenExtensions` with `s`

Comment (rayhan-ab) [reply rollup.config.js]: updated

**Review by aziz-zen-ab — APPROVED** [review]

**Review by aris-fm — APPROVED** [review]

**Review by andrinuryawan — APPROVED** [review]

**Pull request merged**